src/dbt_client/dbtCloudIntegration.ts

```diff
@@ -143,17 +143,22 @@ export class DBTCloudProjectIntegration
     await this.initializePaths();
   }
 
-  async executeSQL(query: string): Promise<QueryExecution> {
+  async executeSQL(query: string, limit: number): Promise<QueryExecution> {
+    const commandArgs = [
+      "show",
+      "--inline",
+      query,
+      "--output",
+      "json",
+      "--log-format",
+      "json",
+    ];
+    // If limit is provided and is a positive number, add the limit arguments
+    if (limit !== undefined && limit > 0) {
+      commandArgs.push("--limit", limit.toString());
+    }
     const showCommand = this.dbtCloudCommand(
-      new DBTCommand("Running sql...", [
-        "show",
-        "--inline",
-        query,
-        "--output",
-        "json",
-        "--log-format",
-        "json",
-      ]),
+      new DBTCommand("Running sql...", commandArgs),
```

Comment:
dbtProject already has

```
if (limit <= 0) {
      window.showErrorMessage("Please enter a positive number for query limit");
      return;
    }
```

You can just add the limit to the parameter, no need for additional validation.

Reply:
Makes sense. Thanks!